drivers/i2c/i2c_mspm0g3xxx.c

```diff
@@ -756,13 +787,22 @@ static int i2c_mspm0g3xxx_init(const struct device *dev)
 	DL_I2C_disableAnalogGlitchFilter((I2C_Regs *)config->base);
 
 	/* Set frequency */
-	uint32_t speed_config = i2c_map_dt_bitrate(config->clock_frequency);
+	uint32_t speed_config = 0;
+	if (config->clock_frequency == I2C_BITRATE_STANDARD ||
+		config->clock_frequency == I2C_BITRATE_FAST) {
+		speed_config = i2c_map_dt_bitrate(config->clock_frequency);
+	} else {
+		/* Device tree specified speed */
+		speed_config = I2C_SPEED_SET(I2C_SPEED_DT);
+	}
```

Comment:
I'd probably flip the conditional so that you treat specifically `I2C_SPEED_DT` as the special case and otherwise call the "default" handling.

Reply:
I can change this to a switch case to highlight what you seem to be considering.

Reply:
That is also okay